src/Aquifer.Public.API/Endpoints/Bibles/List/Validator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using FastEndpoints;
+using FluentValidation;
+
+namespace Aquifer.Public.API.Endpoints.Bibles.List;
+
+public class Validator : Validator<Request>
+{
+    public Validator()
+    {
+        RuleFor(x => x.LanguageId)
+            .Empty()
+            .When(x => x.LanguageCode is not null)
+            .WithMessage(x => "'{PropertyName}' must be empty when 'Language Code' is specified.");
+        RuleFor(x => x.LanguageCode)
+            .Empty()
+            .When(x => x.LanguageId is not null)
+            .WithMessage(x => "'{PropertyName}' must be empty when 'Language Id' is specified.");
+
+        RuleFor(x => x.LanguageId)
+            .InclusiveBetween(1, 32);
```

Comment:
I copied this from [here](https://github.com/BiblioNexusStudio/aquifer-server/blob/0f9cb13ccf06e6ba1a82e5131a3acd0fa0cfef04/src/Aquifer.Public.API/Endpoints/Resources/Updates/List/Validator.cs#L11).  That said, I'm curious if fluent validation supports any form of DB checks.  Could I, say, fetch all languages from the database, cache the maximum ID, and use that ID in the validator?  Are there examples of doing things like that with DI into the validator class?

Context: I'm used to validation logic sitting at the top of my endpoint's definition where you can intermix both basic request correctness checks with data validation of request parameters.

Reply:
You can inject singletons into the validator, but I'm hesitant to add extra db calls without good reason. In a case like this, the chances of someone spamming us with bad language ids is extremely low, so sometimes having some pointless hits to the database because they send a language id higher than what we support seems like far less churn on the database than having to go and check for the list of language ids on _every_ request.

Reply:
> on every request

That's why I mentioned caching it. 😄 

In general I agree that this is fine as is.  I was asking more for the theory behind validation.

My main concern with putting `32` into the validation is that now if anyone adds another language in the DB they have to go find all such language ID validation (I copy/pasted this from somewhere else which copy/pasted it from another place) and update it to `33`.  Or more likely they will forget and the API won't be able to return that new language information until a user submits a bug report.